explorer/mempool/connection_pool.go

```diff
@@ -91,7 +119,8 @@ func (cp *connectionPool) addConnection() error {
 	return nil
 }
 
-func (cp *connectionPool) resetConnection(wsConn *websocketConnection) {
+// resetConnection closes and removes the given connection from the pool, and adds a new one
+func (cp *connectionPool) resetConnection(wsConn *websocketConnection) error {
```

Comment:
I think we have race condition in here since we the lock is released between deletion and recreation

  ```
func (cp *connectionPool) resetConnection(wsConn *websocketConnection) error {
      cp.mu.Lock()
      wsConn.conn.Close()
      delete(cp.connections, wsConn.id)
      delete(cp.connectionsByAddress, wsConn.address.get())
      cp.noMoreConnections = false
      cp.mu.Unlock()  // ← LOCK RELEASED HERE

      return cp.addConnection()  // ← NEW lock acquired inside
  }
```

Reply:
mmm i don't think so? we don't really care which conn arrives to addConnection when it is reset, it is just enough one is handled,  the others are going to be queued and solved once the connectivity issues are solved. Re: https://github.com/arkade-os/go-sdk/pull/70#discussion_r2613733045